--- ./node-v0.4.9/wscript.orig	2011-07-21 13:41:56.000000000 -0500
+++ ./node-v0.4.9/wscript	2011-07-21 14:04:21.000000000 -0500
@@ -354,6 +354,7 @@
     conf.env['DEST_CPU'] = canonical_cpu_type(conf.env['DEST_CPU'])
 
   conf.check(lib='rt', uselib_store='RT')
+  conf.check(lib='z', uselib_store='Z')
 
   if sys.platform.startswith("sunos"):
     if not conf.check(lib='socket', uselib_store="SOCKET"):
@@ -423,6 +424,7 @@
 
 
   conf.define("HAVE_CONFIG_H", 1)
+  conf.define("FULLSTATIC", 1)
 
   if sys.platform.startswith("sunos"):
     conf.env.append_value ('CCFLAGS', '-threads')
@@ -798,7 +800,7 @@
   node = bld.new_task_gen("cxx", product_type)
   node.name         = "node"
   node.target       = "node"
-  node.uselib = 'RT EV OPENSSL CARES EXECINFO DL KVM SOCKET NSL KSTAT UTIL OPROFILE'
+  node.uselib = 'RT EV OPENSSL CARES EXECINFO DL KVM SOCKET NSL KSTAT UTIL OPROFILE Z'
   node.add_objects = 'eio http_parser'
   if product_type_is_lib:
     node.install_path = '${LIBDIR}'
